### PR TITLE
Add helper methods to get the release stage

### PR DESCRIPTION
**Pull request by Azurelol**
## Description

Add helper methods to get the release stage (dev, rc, snapshot, final). This is useful for writing predicates and the like.

## Changes
* ![ADD] Providers in the extension to get the current release stage resolved by the plugin.
 
[NEW]:      https://resources.atlas.wooga.com/icons/icon_new.svg "New"
[ADD]:      https://resources.atlas.wooga.com/icons/icon_add.svg "Add"
[IMPROVE]:  https://resources.atlas.wooga.com/icons/icon_improve.svg "Improve"
[CHANGE]:   https://resources.atlas.wooga.com/icons/icon_change.svg "Change"
[FIX]:      https://resources.atlas.wooga.com/icons/icon_fix.svg "Fix"
[UPDATE]:   https://resources.atlas.wooga.com/icons/icon_update.svg "Update"

[BREAK]:    https://resources.atlas.wooga.com/icons/icon_break.svg "Remove"
[REMOVE]:   https://resources.atlas.wooga.com/icons/icon_remove.svg "Remove"
[IOS]:      https://resources.atlas.wooga.com/icons/icon_iOS.svg "iOS"
[ANDROID]:  https://resources.atlas.wooga.com/icons/icon_android.svg "Android"
[WEBGL]:    https://resources.atlas.wooga.com/icons/icon_webGL.svg "WebGL"
[GRADLE]:   https://resources.atlas.wooga.com/icons/icon_gradle.svg "GRADLE"
[UNITY]:    https://resources.atlas.wooga.com/icons/icon_unity.svg "Unity"
[LINUX]:    https://resources.atlas.wooga.com/icons/icon_linux.svg "Linux"
[WIN]:      https://resources.atlas.wooga.com/icons/icon_windows.svg "Windows"
[MACOS]:    https://resources.atlas.wooga.com/icons/icon_iOS.svg "macOS"

### CODE REVIEW

File: src/main/groovy/wooga/gradle/version/VersionPluginConventions.groovy [hunk 1]
```diff
@@ -25,37 +25,6 @@ class VersionPluginConventions {
...
-
-    static final String VERSION_STAGE_OPTION = "versionBuilder.stage"
-    static final String VERSION_STAGE_ENV_VAR = "VERSION_BUILDER_STAGE"
-
-    static final String RELEASE_BRANCH_PATTERN_OPTION = "versionBuilder.releaseBranchPattern"
-    static final String RELEASE_BRANCH_PATTERN_ENV_VAR = "VERSION_BUILDER_RELEASE_BRANCH_PATTERN"
-
-    static final String MAIN_BRANCH_PATTERN_OPTION = "versionBuilder.mainBranchPattern"
-    static final String MAIN_BRANCH_PATTERN_ENV_VAR = "VERSION_BUILDER_MAIN_BRANCH_PATTERN"
-
```
Comment (Larusso) [inline src/main/groovy/wooga/gradle/version/VersionPluginConventions.groovy:58]: All these values are no longer in use?

File: src/main/groovy/wooga/gradle/version/internal/release/opinion/Strategies.groovy [hunk 2]
```diff
@@ -30,10 +31,7 @@ import org.gradle.api.GradleException
 /**
  * Opinionated sample strategies. These can either be used as-is or as an
  * example for others.
- * @see wooga.gradle.version.internal.release.base.VersionStrategy
- * @see SemVerStrategy
- * @see wooga.gradle.version.internal.release.semver.SemVerStrategyState
- * @see PartialSemVerStrategy
+ * @see wooga.gradle.version.internal.release.base.VersionStrategy* @see SemVerStrategy* @see wooga.gradle.version.internal.release.semver.SemVerStrategyState* @see PartialSemVerStrategy
```
Comment (Larusso) [inline src/main/groovy/wooga/gradle/version/internal/release/opinion/Strategies.groovy:34]: Be carefull when reformating code that the docs are not folded.

**Review by Larusso — CHANGES_REQUESTED** [review]

File: src/main/groovy/wooga/gradle/version/VersionPluginConventions.groovy (hunk 1 above)
Comment (Azurelol) [reply src/main/groovy/wooga/gradle/version/VersionPluginConventions.groovy:58]: They were refactored into the PropertyLookup classes on a previous PR; had forgotten to delete the old fields.

File: src/main/groovy/wooga/gradle/version/internal/release/opinion/Strategies.groovy (hunk 2 above)
Comment (Azurelol) [reply src/main/groovy/wooga/gradle/version/internal/release/opinion/Strategies.groovy:34]: My bad.

File: src/main/groovy/wooga/gradle/version/VersionPluginConventions.groovy (hunk 1 above)
Comment (Larusso) [reply src/main/groovy/wooga/gradle/version/VersionPluginConventions.groovy:58]: ok

**Review by Larusso — APPROVED** [review]

**Pull request merged**